packages/chat/src/views/View.js

```diff
@@ -106,21 +106,38 @@ const Textarea = ({ state, dispatch }) => {
       <textarea
         className='w-full' rows='5' readOnly={!!state.loading}
         value={value(state, 'content')} onChange={change.bind(null, dispatch, 'content')}
+        onKeyDown={handleKeys.bind(null, state, dispatch)}
       />
     )
   }
   return (
     <textarea
       className='w-full' rows='5' value={value(state, 'content')} onChange={change.bind(null, dispatch, 'content')}
-      onKeyDown={e=>{handleMetaEnter(e)}}
+      onKeyDown={handleKeys.bind(null, state, dispatch)}
     />
   )
 }
 
-const handleMetaEnter = e => {
-  if(e.key === 'Enter' && (e.metaKey || e.ctrlKey)){
+const handleKeys = async (state, dispatch, e) => {
+  if (e.key === 'Enter' && (e.metaKey || e.ctrlKey)) {
     e.preventDefault()
-    e.target.form.requestSubmit()
+    await sendMessage(state, dispatch, e)
+  } else if (e.key === 'ArrowUp' && (e.metaKey || e.ctrlKey) && state && state.channels) {
+    e.preventDefault()
+    const _channels = Object.values(state.channels)
+    const _current = _channels.find(c => {
+      return state.active === c.data.name
+    })
+    const _index = (_channels.indexOf(_current) - 1 < 0 ? _channels.length - 1 : _channels.indexOf(_current) - 1)
+    await changeChannel(state, dispatch, _channels[_index])
+  } else if (e.key === 'ArrowDown' && (e.metaKey || e.ctrlKey) && state && state.channels) {
+    e.preventDefault()
+    const _channels = Object.values(state.channels)
```

Comment:
instead of duplicating the logic, let's pull this out into it's own function.

additionally, if we're thinking about the problem we're trying to solve it's really pretending we have an array of channels at some index and we want to either move up or down the index and wrap around on either end.

a little bit of googling brings us to this excellent SO -
https://stackoverflow.com/a/45397704

```
const switchChannel = (state, step) => {
  const { active, channels } = state
  const cnt = Object.keys(channels).length
  const index = Object.values(channels).reduce((acc, {data: { name }, i) => name === active ? i : acc, -1) + step

  // https://stackoverflow.com/a/45397704
  const newIndex = (index % cnt + cnt) % cnt
  const channel = Object.values(channels)[newIndex]
  return channel
}

const STEPS = {'ArrowDown': -1, 'ArrowUp': 1}

const handleKeys = async (state, dispatch, e) => {
  // bail early if meta/ctrl key not pressed
  if (!(e.metaKey || e.ctrlKey)) {
    return
  }

  // switch channels
  if (Object.keys(STEPS).includes(e.key)) {
    const step = STEPS[e.key]
    const channel = switchChannel(state, step)
    dispatch({ type: 'active', payload: channel.data.name })
    dispatch({ type: 'channelId', payload: channel.id })
    return
  }

  // handle other shortcuts like send etc.
}

```

Reply:
btw I haven't tested this code but you should get the general idea.